pwr.md

```diff
@@ -0,0 +1,144 @@
+>Available frequencies are between `uncore_hw_max` and `uncore_hw_min`.
+
+```python
+core.min_freq = core.lowest_freq
+core.max_freq = core.highest_freq
+
+cpu.uncore_max_freq = cpu.uncore_hw_max
+cpu.uncore_min_freq = cpu.uncore_hw_min
+```
+
+### Committing 
+Modification of the power settings of a system is done by altering the core or CPU characteristics, as shown above, 
+then finalizing with the `commit()` function call, this will commit your changes and write to relevant MSRs or sysfs entries.
+```python 
+cores = pwr.get_cores()
+
+for core in cores:
+    core.min_freq = core.base_freq
+    core.max_freq = core.highest_freq
+    core.commit()
+```
+
+
+### Pre-set Profiles
+When an application is modifying the desired min and max core frequencies, pre-set configurations can also be applied, these will overwrite current configurations and commit the pre-sets.  
+* `minimum`:            Set all cores minimum to 800Mhz and maximum to 800Mhz.
+* `maximum`:            Set all cores minimum to 3900Mhz and maximum to 3900Mhz.
+* `base`:               Set all cores minimum to 2300Mhz and maximum to 2300Mhz.
+* `default`:            Set all cores minimum to 800Mhz and maximum to 3900Mhz.
+* `no_turbo`:           Set all cores minimum to 800Mhz and maximum to 2300Mhz.
+* `sst_bf_base`:        Set high priority cores minimum and maximum to 2700Mhz, normal priority cores minimum and maximum to 2100Mhz.
+* `sst_bf_high_turbo`:  Set high priority cores minimum to 2700Mhz and maximum to 3900Mhz, normal priority cores minimum to 800Mhz and maximum to 2100Mhz.
+* `sst_bf_low_turbo`:   Set high priority cores minimum to 2700Mhz and maximum to 2700Mhz, normal priority cores minimum to 800Mhz and maximum to 3900Mhz.
+>Specific frequencies will depend on system and configuration.
+
+```python
+For c in cores:
+    c.commit("sst_bf_base") # Configure cores with the SST_BF configuration
+```
+
+## Refreshing cpu stats
+On library initialization the systems capabilities and stats are retrieved.
+CPU stats can become out of date, such as `curr_freq` or `sst_bf_configured`. These can be refreshed in with `refresh_stats()`.
+It is *advised* that CPU and core objects be updated with `refresh_stats()` before requesting object data.
+
+core.refresh_stats() will update:
+* `curr_freq`             
+* `min_freq`                
+* `max_freq`                
+* `epp`
+                
+cpu.refresh_stats() will update:
+* `uncore_freq`
+* `uncore_max_freq` 
+* `uncore_min_freq`
+* `sst_bf_configured`
+
+```python
+for c in cores:
+    c.refresh_stats()
+    if c.sst_bf_configured:
+        c.commit("sst_bf_high_turbo")
+```
+
+## Object Referencing
+On object list creation, with `get_cores()/get_cpus()`, *both* object lists are created from *either* call.
+This is due to fact that both objects reference each other, the list of core objects can be accessed through the CPU and the CPU object of a core can be accessed through a core.
```

Comment:
Similar here: internal details aren't needed. What _does_ matter is getting across that CPU and core objects will reference each other, and it is possible to get CPU from core and vice versa.

Reply:
from what I can tell, that is what is line is saying.

Reply:
Yes, among 10 other points it makes that aren't pertinent :)